src/main.ts

```diff
@@ -0,0 +1,34 @@
+import { Actor,log } from 'apify';
+
+import { generateTweetFromWebContent } from './ai-utils.js';
+import { validateInput } from './input.js';
+import { searchQuery } from './rag-web-search.js';
+import type { Input } from './types.js';
+
+try {
+    await Actor.init();
+
+    const originalInput = await Actor.getInput<Input>() ?? {} as Input;
```

Comment:
We need to avoid using TS casts as much as possible.
I'd rather do 👇 

```suggestion
    const originalInput = await Actor.getInput<Input>();
    if (!originalInput) {
       throw new Error("Actor input can not be empty!")
    }
```

Reply:
Solved by commit 9beae144db531b194796cf43adb94bc1d6fdacb2